serializations.md

```diff
@@ -644,6 +646,22 @@ The payload is a serialized signed channel off-chain transaction and can not be
 ]
 ```
 
+version 2:
+```
+[ <channel_id>      :: id()
+, <from_id>         :: id()
+, <payload>         :: binary()
+, <round>           :: int()
+, <update>          :: binary()
+, <state_hash>      :: binary()
+, <offchain_trees>  :: trees()
+, <ttl>             :: int()
+, <fee>             :: int()
+, <nonce>           :: int()
+, <block_hash>       :: binary()
```

Comment:
I see block hash here and, if payload is off-chain version 2, also in payload. Any constraints or interactions? Which one is picked? (Maybe this is documented somewhere else?)

Reply:
The off-chain tx in payload is a proof for agreement that the state it brings was a valid state at one point of time - it is co-signed. We use it for 2 things:
1. `round` - defines the next round to be produced
2. `state_hash` proves that the provided `offchain_trees` matche the expectations: we check the root hash against the `state_hash` of the payload

The `block_hash` in the payload, if any, is discarded (they could have used any block to base their old update). All we care here is the new update we're about ot produce on-chain, thus the force progress `block_hash`